### ISSUE TITLE
Trouble testing XEP-0198

### ISSUE THREAD

**Issue by detrout** [noonien-d/telepathy-gabble-old#2]
I was trying to test support, and I'm not sure if this should work or not.

I disabled all of my other jabber clients except for the one running your branch, I went to a different computer, prepared to send a message, pulled the ethernet cable from the test computer and in about 3-5 seconds sent the message from the other computer. I then reconnected the Ethernet cable and hoped to see the undelivered message.

Unfortunately I didn't seem to. I'm currently using kde-telepathy and haven't implemented any client side support.

**Comment by noonien-d** [comment]
Did you check the server side support for xep-0198? Some inplementations don't resent the message, but send an error to the sender instead  (which may not even be displayed, depending on the client).

**Comment by detrout** [comment]
Conversations says XEP-0198 is enabled on my server. Though it looks like I had an old version of mod_smacks. I had 1416:7ddb522d9b28 of the prosody community modules from May 2014 installed.

I'll try again tonight after updating the prosody modules.

**Comment by noonien-d** [comment]
Mod_smacks will only resend the messages in the case of "fast resumption", which is not supported by this gabble patchset. 
If your client gets finally disconnected by timeout or crash (or doesn't support resumption), the message will be reported as undelivered [1]. 

You'll also need the "mod_smack_offline" module [2] to enable resending of messages. Note that it will only work if the recipient uses only one client at once.

[1] http://modules.prosody.im/mod_smacks.html#details
[2] http://modules.prosody.im/mod_smacks_offline.html

**Comment by detrout** [comment]
I figured out how to get more gabble logging turned on from gabble and I can see the sm:3 h messages. I enabled smacks_offline and I may have gotten an undelivered message. My test wasn't very easy to replicate.

I'll run your patches for a bit longer and then report back to the freedesktop bug. (I have contacts in KDE Telepathy who hopefully can get things committed to freedesktop.)

**Comment by noonien-d** [comment]
For everyone else wanting to check if stream management is working, start gabble with debug output enabled:

```
G_MESSAGES_DEBUG=all WOCKY_DEBUG=xmpp ./telepathy-gabble
```

Another possibility for empathy users is to run "empathy-debugger" with the "gabble (jabber)" filter selected. 

If stream management is enabled and working, you'll see stanzas like:

```
* r xmlns='urn:xmpp:sm:3'
* a xmlns='urn:xmpp:sm:3' h='4'
```

@detrout 
Thanks for your debugging, I'm looking forward for your feedback!

**Comment by detrout** [comment]
After running for a while I got the following, and telepathy-gabble exited. I'll try some more to see if I can localize it.

(telepathy-gabble:24511): wocky-WARNING **: wocky_sm_pop_unacked_stanza: sm == NULL

(telepathy-gabble:24511): wocky-WARNING **: wocky_sm_pop_unacked_stanza

(telepathy-gabble:24511): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

**Comment by noonien-d** [comment]
This should be fixed by noonien-d/wocky@b799dc266b2d8d369ac6caff6416643c3b498feb
